books/free-programming-books-subjects.md

```diff
@@ -649,6 +650,7 @@ Books that cover a specific programming language can be found in the  [BY PROGRA
 * [How to write a simple operating system in assembly language](http://mikeos.sourceforge.net/write-your-own-os.html) - Mike Saunders (HTML)
 * [Linux From Scratch](https://www.linuxfromscratch.org/lfs/view/stable/) - Gerard Beekmans, Bruce Dubbs, Ken Moffat, Pierre Labastie, et al. (HTML, [PDF, downloads...](https://www.linuxfromscratch.org/lfs/downloads/stable/))
 * [Making Servers Work: A Practical Guide to Linux System Administration](https://www.digitalocean.com/community/books/sysadmin-ebook-making-servers-work) - Jamon Camisso (PDF, EPUB)
+* [Operating System concepts](https://drive.uqu.edu.sa/_/mskhayat/files/MySubjects/2017SS%20Operating%20Systems/Abraham%20Silberschatz-Operating%20System%20Concepts%20(9th,2012_12).pdf) - Avi Silberschatz, Greg Gagne, and Peter Baer Galvin (PDF)
```

Comment:
is it really free? no illegal resources

Reply:
Remove illegal content

Publisher website: https://www.wiley.com/en-us/Operating+System+Concepts%2C+10th+Edition-p-9781119320913

```suggestion
```